lib/puppet/provider/mysql.rb

```diff
@@ -24,7 +24,7 @@ def self.mysqld_type
     mysqld_version_string.scan(/mariadb/i) { return "mariadb" }
     mysqld_version_string.scan(/\s\(mysql/i) { return "mysql" }
     mysqld_version_string.scan(/\s\(percona/i) { return "percona" }
-    nil
+    return "mysql"
```

Comment:
That sounds OK in theory, but it only works if it is caught where it is used. It is most definitely not caught in the mysql_user provider.

I feel that mariadb and percona are the special cases that need to be trapped, not mysql. If there is a "mysqld" executable, it should be assumed to actually be mysql unless a special case is detected.